daemon.c

```diff
@@ -37,7 +37,7 @@ daemonize(char * pid_file) {
 		exit(EXIT_SUCCESS);
 	umask(0);
 	chdir("/");
-	for (fd = sysconf(_SC_OPEN_MAX); fd > 0; fd--)
+	for (fd = sysconf(_SC_OPEN_MAX)-1; fd >= 0; fd--)
```

Comment:
I understand why fd should be >= 0, but what is the purpose of -1 in the loop? we count down from max fds that could be opened down to 0.

Reply:
Since we are starting the fd count from 0, the max valid fd should be 1 less than the max count

Reply:
I guess, the idea is that actual fd would be less than count.
Looks fine.